solidity/contracts/EthBonding.sol

```diff
@@ -0,0 +1,164 @@
+/**
+▓▓▌ ▓▓ ▐▓▓ ▓▓▓▓▓▓▓▓▓▓▌▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▄
+▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▌▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+  ▓▓▓▓▓▓    ▓▓▓▓▓▓▓▀    ▐▓▓▓▓▓▓    ▐▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌   ▐▓▓▓▓▓▓
+  ▓▓▓▓▓▓▄▄▓▓▓▓▓▓▓▀      ▐▓▓▓▓▓▓▄▄▄▄         ▓▓▓▓▓▓▄▄▄▄         ▐▓▓▓▓▓▌   ▐▓▓▓▓▓▓
+  ▓▓▓▓▓▓▓▓▓▓▓▓▓▀        ▐▓▓▓▓▓▓▓▓▓▓         ▓▓▓▓▓▓▓▓▓▓▌        ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+  ▓▓▓▓▓▓▀▀▓▓▓▓▓▓▄       ▐▓▓▓▓▓▓▀▀▀▀         ▓▓▓▓▓▓▀▀▀▀         ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▀
+  ▓▓▓▓▓▓   ▀▓▓▓▓▓▓▄     ▐▓▓▓▓▓▓     ▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌
+▓▓▓▓▓▓▓▓▓▓ █▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+
+                           Trust math, not hardware.
+*/
+
+pragma solidity 0.5.17;
+
+import "./AbstractBonding.sol";
+
+import "@keep-network/keep-core/contracts/Authorizations.sol";
+import "@keep-network/keep-core/contracts/KeepRegistry.sol";
+import "@keep-network/keep-core/contracts/StakeDelegatable.sol";
+
+import "@keep-network/sortition-pools/contracts/api/IFullyBackedBonding.sol";
+
+/// @title ETH Bonding
+/// @notice Contract holding deposits and delegations for ETH-only keeps'
+/// operators. An owner of the ETH can delegate ETH to an operator. The value
+/// of ETH the owner is willing to delegate should be deposited for the given
+/// operator.
+contract EthBonding is
+    AbstractBonding,
+    Authorizations,
+    StakeDelegatable,
+    IFullyBackedBonding
+{
+    event Delegated(address indexed owner, address indexed operator);
+
+    event OperatorDelegated(
+        address indexed operator,
+        address indexed beneficiary,
+        address indexed authorizer
+    );
+
+    // TODO: Decide on the final value and if we want a setter for it.
+    uint256 public constant MINIMUM_DELEGATION_DEPOSIT = 12345;
+
+    uint256 public initializationPeriod;
+
+    /// @notice Initializes Keep Bonding contract.
+    /// @param _keepRegistry Keep Registry contract address.
+    /// @param _initializationPeriod To avoid certain attacks on group selection,
+    /// recently delegated operators must wait for a specific period of time
+    /// before being eligible for group selection.
+    constructor(KeepRegistry _keepRegistry, uint256 _initializationPeriod)
+        public
+        AbstractBonding(_keepRegistry)
+        Authorizations(_keepRegistry)
+    {
+        initializationPeriod = _initializationPeriod;
+    }
+
+    /// @notice Registers delegation details. The function is used to register
+    /// addresses of operator, beneficiary and authorizer for a delegation from
+    /// the caller.
+    /// The function requires ETH to be submitted in the call as a protection
+    /// against attacks blocking operators. The value should be at least equal
+    /// to the minimum delegation deposit. Whole amount is deposited as operator's
+    /// unbonded value for the future bonding.
```

Comment:
The fact we are preventing operator griefing this way sounds to me like a candidate for `/// @dev` detail.